apps/heureka/src/components/Services/common/ServiceImageVersions/ServiceImageVersions.tsx

```diff
@@ -0,0 +1,179 @@
+import { ServiceType } from "../../Services"
+
+export type ServiceImageVersion = {
+  imageName: string
+  imageVersion: string
+  imageTag: string
+  issueCounts: {
+    critical: number
+    high: number
+    medium: number
+    low: number
+    none: number
+  }
+  serviceName: string
+  totalCount?: number
+}
+
+type ServiceImageVersionsProps = {
+  service: ServiceType
+  showDetailsButtons?: boolean
+}
+
+export const ServiceImageVersions = ({ service, showDetailsButtons }: ServiceImageVersionsProps) => {
+  const dispatch = useDispatch()
+  const { name: serviceName } = service
+  const { loading, imageVersions, error, totalNumberOfPages, currentPage, goToPage, totalCount } =
+    useFetchServiceImageVersions({
+      serviceCcrn: serviceName || "",
+      pageSize: 10,
+    })
+
+  const formattedImageVersions = imageVersions.map((version) => ({
+    imageName: version.ccrn,
+    imageVersion: version.version,
+    imageTag: version.tag,
+    issueCounts: version.issueCounts,
+    serviceName,
+  }))
+
+  const columnCount = showDetailsButtons ? 7 : 6
+
+  const showServiceDetails = useCallback(
+    ({ service, imageVersion }: { service: ServiceType; imageVersion?: ServiceImageVersion }) => {
+      dispatch({
+        type: ActionType.SelectView,
+        payload: {
+          viewId: UserView.ServiceDetails,
+          params: {
+            service,
+            imageVersion,
+          },
+        },
+      })
+    },
+    []
+  )
+
+  return (
+    <Stack gap="4" direction="vertical" className="w-full">
+      <Stack distribution="between" alignment="center" className="w-full">
+        {showDetailsButtons ? (
+          <>
+            <span>{totalCount} image versions in service</span>
+            <Button variant="primary" size="small" onClick={() => showServiceDetails({ service })}>
+              Full Details
+            </Button>
+          </>
+        ) : (
+          <ContentHeading>Service Image Versions ({totalCount})</ContentHeading>
+        )}
+      </Stack>
+      <DataGrid columns={columnCount}>
+        <DataGridRow>
+          <DataGridHeadCell>Image Name</DataGridHeadCell>
+          <DataGridHeadCell>Tag</DataGridHeadCell>
+          <DataGridHeadCell>Critical</DataGridHeadCell>
+          <DataGridHeadCell>High</DataGridHeadCell>
+          <DataGridHeadCell>Medium</DataGridHeadCell>
+          <DataGridHeadCell>Low</DataGridHeadCell>
+          {showDetailsButtons && <DataGridHeadCell>Actions</DataGridHeadCell>}
+        </DataGridRow>
+        {loading ? (
+          <EmptyDataGridRow colSpan={columnCount}>Loading...</EmptyDataGridRow>
+        ) : imageVersions?.length === 0 && !error ? (
+          <EmptyDataGridRow colSpan={columnCount}>No image versions available.</EmptyDataGridRow>
+        ) : (
+          formattedImageVersions.map((version, index) => (
+            <DataGridRow key={index}>
+              <DataGridCell className="break-all overflow-hidden">
```

Comment:
Can we make a global class for it with a description so it is clear why we have here custom styles :)